src/core/execute-serialize.c

```diff
@@ -19,6 +19,513 @@
+                        return r;
+        }
+
+        if (p->n_socket_fds + p->n_storage_fds > 0) {
+                _cleanup_free_ char *serialized_fds = NULL;
+
+                if (!p->fds)
+                        return -EINVAL;
+
+                for (size_t i = 0; i < p->n_socket_fds + p->n_storage_fds; ++i) {
+                        int copy = -EBADF;
+
+                        if (p->fds[i] >= 0) {
+                                copy = fdset_put_dup(fds, p->fds[i]);
+                                if (copy < 0)
+                                        return copy;
+                        }
+
+                        if (strextendf(&serialized_fds, "%d ", copy) < 0)
+                                return -ENOMEM;
+                }
+
+                r = serialize_item(f, "exec-parameters-fds", serialized_fds);
+                if (r < 0)
+                        return r;
+        }
+
+        r = serialize_strv(f, "exec-parameters-fd-names", p->fd_names);
+        if (r < 0)
+                return r;
+
+        if (p->flags > 0) {
+                r = serialize_item_format(f, "exec-parameters-flags", "%u", p->flags);
+                if (r < 0)
+                        return r;
+        }
+
+        r = serialize_bool_elide(f, "exec-parameters-selinux-context-net", p->selinux_context_net);
+        if (r < 0)
+                return r;
+
+        if (p->cgroup_supported > 0) {
+                r = serialize_item_format(f, "exec-parameters-cgroup-supported", "%u", p->cgroup_supported);
+                if (r < 0)
+                        return r;
+        }
+
+        r = serialize_item(f, "exec-parameters-cgroup-path", p->cgroup_path);
+        if (r < 0)
+                return r;
+
+        r = serialize_item_format(f, "exec-parameters-cgroup-id", "%" PRIu64, p->cgroup_id);
+        if (r < 0)
+                return r;
+
+        r = serialize_strv(f, "exec-parameters-prefix", p->prefix);
+        if (r < 0)
+                return r;
+
+        r = serialize_item(f, "exec-parameters-received-credentials-directory", p->received_credentials_directory);
+        if (r < 0)
+                return r;
+
+        r = serialize_item(f, "exec-parameters-received-encrypted-credentials-directory", p->received_encrypted_credentials_directory);
+        if (r < 0)
+                return r;
+
+        r = serialize_item(f, "exec-parameters-confirm-spawn", p->confirm_spawn);
+        if (r < 0)
+                return r;
+
+        r = serialize_bool_elide(f, "exec-parameters-shall-confirm-spawn", p->shall_confirm_spawn);
+        if (r < 0)
+                return r;
+
+        if (p->watchdog_usec > 0) {
+                r = serialize_usec(f, "exec-parameters-watchdog-usec", p->watchdog_usec);
+                if (r < 0)
+                        return r;
+        }
+
+        if (p->idle_pipe) {
+                _cleanup_free_ char *serialized_fds = NULL;
+
+                for (size_t i = 0; i < 4; ++i) {
```

Comment:
```
../src/core/execute-serialize.c: In function ‘exec_parameters_serialize’:
../src/fundamental/macro-fundamental.h:65:26: warning: division ‘sizeof (int * const) / sizeof (int)’ does not compute the number of array elements [-Wsizeof-pointer-div]
   65 |                 sizeof(x)/sizeof((x)[0]),                               \
      |                          ^
../src/core/execute-serialize.c:1452:40: note: in expansion of macro ‘ELEMENTSOF’
 1452 |                 for (size_t i = 0; i < ELEMENTSOF(p->idle_pipe); ++i) {
      |                                        ^~~~~~~~~~
../src/fundamental/macro-fundamental.h:57:19: error: void value not ignored as it ought to be
   57 | #  define VOID_0 ((void)0)
      |                   ^
../src/fundamental/macro-fundamental.h:66:17: note: in expansion of macro ‘VOID_0’
   66 |                 VOID_0))
      |                 ^~~~~~
../src/core/execute-serialize.c:1452:40: note: in expansion of macro ‘ELEMENTSOF’
 1452 |                 for (size_t i = 0; i < ELEMENTSOF(p->idle_pipe); ++i) {
```

I think this is because it's not a fixed array, it's a pointer